python/paddle/fluid/optimizer.py

```diff
@@ -275,15 +275,26 @@ def _create_optimization_pass(self, parameters_and_grads):
         self._create_global_learning_rate()
 
         optimize_ops = []
-        for param_and_grad in parameters_and_grads:
-            if param_and_grad[1] is None:
-                continue
-            with param_and_grad[0].block.program._optimized_guard(
-                    param_and_grad), name_scope("optimizer"):
-                if param_and_grad[0].trainable is True:
-                    optimize_op = self._append_optimize_op(global_block,
-                                                           param_and_grad)
-                    optimize_ops.append(optimize_op)
+        if framework._in_dygraph_mode():
+            for param_and_grad in parameters_and_grads:
+                if param_and_grad[1] is None:
+                    continue
+                with param_and_grad[0].block.program._optimized_guard(
+                        param_and_grad):
+                    if param_and_grad[0].trainable is True:
+                        optimize_op = self._append_optimize_op(global_block,
+                                                               param_and_grad)
+                        optimize_ops.append(optimize_op)
+        else:
+            for param_and_grad in parameters_and_grads:
+                if param_and_grad[1] is None:
+                    continue
+                with param_and_grad[0].block.program._optimized_guard(
+                        param_and_grad), name_scope("optimizer"):
```

Comment:
Why no need `NameScope` in dygraph mode?
If `NameScope` is not suitable with DyGraph mode, maybe we can add an assert in the `NameScope()` API.

Reply:
because ```NameScope With Guard``` will create a new ```NameScope``` each time, so python will take a long time to do GC after few batchs, actually ```name_scope``` didn't take effect in DyGraph mode